File: server-spring-boot/src/main/java/com/finflow/server/entity/Customer.java
package com.finflow.server.entity;

import java.util.List;

import jakarta.persistence.*;
import lombok.*;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Customer {

   @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String email;
    private String password;
    private Double balance;
    private Double monthlySalary;
    private Double monthlyBudget;

    @OneToMany(mappedBy = "customer")
    private List<Expense> expenses;

    @OneToMany(mappedBy = "customer")
    private List<LoanRequest> loanRequests;

    @OneToMany(mappedBy = "customer")
    private List<Invoice> invoices;
}

File: server-dot-net/Model/Customer.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace server_dot_net.Model
{
    [Table("customer")]
    public class Customer
    {
        [Key]
        [Column("id")]
        public long Id { get; set; }

        [Required]
        [Column("name")]
        public string Name { get; set; } = string.Empty;

        [Required]
        [Column("email")]
        public string Email { get; set; } = string.Empty;

        [Required]
        [Column("password")]
        public string Password { get; set; } = string.Empty;

        [Column("balance")]
        public double Balance { get; set; }

        [Column("monthly_salary")]
        public double MonthlySalary { get; set; }

        [Column("monthly_budget")]
        public double MonthlyBudget { get; set; }
    }
}
